### DISCUSSION TITLE
Some details about the turbulence closures

### DISCUSSION THREAD

**Discussion by tomchor** [CliMA/Oceananigans.jl#1321] [Computational science!] [answered]
Hi, guys, I was going through the code and I have a few questions regarding the turbulence closures:

- From the code it seems that the turbulence closures always use the full modeled velocities, right? In other words:
  - It doesn't include any background flows.
  - If I want to use only a fluctuation around a user-defined mean (which apparently can improve performance of some models), I can't do it.
  Am I right about these statements?
- I saw that there are more turbulent closures included in `turbulence_closure_implementations` than what's documented. Are these not fully tested?

Thanks!

**Comment by glwagner** [comment, marked as answer]
> It doesn't include any background flows.

Correct. This might be fixable without touching the turbulence closure implementations (which are complicated), but would require some new types (like an abstraction that computes the sum of `velocities.u` and `background_fields.u` when indexed into) and some modification of the time-stepping kernels. Probably not too much typing, but we need to make sure we have a good design.

I think this would be a positive addition to the code. I do think a case would have to have very strongly sheared mean flows with relatively weak turbulence for such a change to have a measureable impact. For AMD, I think such a change might decrease the eddy diffusivity, since it would mostly affect the demoninator. For Smagorinsky-Lilly it would increase the eddy diffusivity (but this is unlikely to be what one wants... ?)

~~The tricky thing is that, if significant turbulent diffusion of a background velocity profile occurs, I'm not 100% sure its appropriate to prescribe it. A better modeling approach for this case might be to relax the velocity field very weakly to the desired mean profile. (It'd be nice to be able to relax _only_ the mean velocity field, but this requires the `model.auxiliaries` concept).~~

EDIT: I realized that the point here is only to include the mean profile in the eddy viscosities / diffusivities. We'd still neglect the turbulent flux associated with the mean velocities.

**Reply by tomchor** [reply]
Thanks for the answer! Btw, I'm not proposing that we add this new feature, I just wanted to know how things worked.

I'm still curious about the other "hidden" closures though. They seem to be ready to be used!

**Reply by glwagner** [reply]
Ah ok. Hmm yes I have been meaning to nuke Rozema for a while so that people aren't tempted 😏 

The [docs](https://clima.github.io/OceananigansDocumentation/stable/model_setup/background_fields/) say

> Notice that the term \nabla \cdot \left ( \bm{U} C \right )∇⋅(UC) is neglected: only the terms describing the advection of resolved tracer by the background velocity field and the advection of background tracer by the resolved velocity field are included. An analgous statement holds for the advection of background momentum by the resolved velocity field. Other possible terms associated with the Coriolis force, buoyancy, turbulence closures, and surface waves acting on background fields are neglected.
